docs/src/main/sphinx/connector/hive.rst

```diff
@@ -984,6 +986,109 @@ Procedures
   Flush Hive metadata cache entries connected with selected partition.
   Procedure requires named parameters to be passed
 
+.. _hive_table_properties:
+
+Table properties
+----------------
+
+Table properties supply or set metadata for the underlying tables. This
+is key for :doc:`/sql/create-table-as` statements. Table properties are passed
+to the connector using a :doc:`WITH </sql/create-table-as>` clause::
+
+  CREATE TABLE tablename 
+  WITH (format='CSV', 
+        csv_escape = '"')
+
+See the :ref:`hive_examples` for more information.
+
+.. list-table:: Hive connector table properties
+  :widths: 20, 60, 20
+  :header-rows: 1
+
+  * - Property name
+    - Description
+    - Default
+  * - ``auto_purge``
+    - Indicates to the configured metastore to perform a purge when a table or 
+      partition is deleted instead of a soft deletion using the trash.
+    -
+  * - ``avro_schema_url``
+    - The URI pointing to :ref:`hive_avro_schema` for the table.
+    -
+  * - ``bucket_count``
+    - The number of buckets to group data into. Only valid if used with
+      ``bucketed_by``.
+    - 0
+  * - ``bucketed_by``
+    - The bucketing column for the storage table. Only valid if used with 
+      ``bucket_count``.
+    -
+  * - ``bucketing_version``
+    - Specifies which Hive bucketing version to use. Valid values are ``1`` 
+      or ``2``.
+    -
+  * - ``csv_escape``
+    - The CSV escape character. Requires CSV format.
+    -
+  * - ``csv_quote``
+    - The CSV quote character. Requires CSV format.
+    -
+  * - ``csv_separator``
+    - The CSV separator character. Requires CSV format.
+    -
+  * - ``external_location``
+    - The URI for an external Hive table on S3, Azure Blob Storage, etc. See the
+      :ref:`hive_examples` for more information.
+    -
+  * - ``format``
+    - The table file format. Valid values include ``ORC``, ``PARQUET``, ``AVRO``, 
+      ``RCBINARY``, ``RCTEXT``, ``SEQUENCEFILE``, ``JSON``, ``TEXTFILE``, and 
+      ``CSV``. The catalog property ``hive.storage-format`` sets the default 
+      value and can change it to a different default.
+    - 
+  * - ``null_format``
+    - The serialization format for ``NULL`` value. Requires TextFile, RCText, 
+      or SequenceFile format.
```

Comment:
Just for information - `null_format` can be set to a string that is then interpreted as a marker for `NULL` values in table files.

e.g. with `null_format = 'NULL_VALUE_MARKER'` and a TextFile table with some line like `a\u0001NULL_VALUE_MARKER\u0001null value` would get read as 3 columns `a`, `NULL` and `null value`.

If we want to document this it should be a separate effort - not part of this PR.

Reply:
@mosabua how should I proceed here?